File: Solutions in Python/Merge_Two_Sorted_List.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def mergeTwoLists(self, list1: Optional[ListNode], list2: Optional[ListNode]) -> Optional[ListNode]:
        vec = []

        # Merge the two lists
        while list1:
            vec.append(list1.val)
            list1 = list1.next

        while list2:
            vec.append(list2.val)
            list2 = list2.next

        vec.sort()

        temp = ListNode(-1)
        head = temp
        for value in vec:
            temp.next = ListNode(value)
            temp = temp.next

        head = head.next
        return head

File: Solutions in Java/Merge_two_sorted_list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode mergeTwoLists(ListNode list1, ListNode list2) {
        ListNode dummy = new ListNode(0); // Dummy node to simplify the merge process
        ListNode current = dummy; // Pointer to build the new list

        // Merge the two lists
        while (list1 != null && list2 != null) {
            if (list1.val <= list2.val) {
                current.next = list1; // Link to the smaller node
                list1 = list1.next; // Move to the next node in list1
            } else {
                current.next = list2; // Link to the smaller node
                list2 = list2.next; // Move to the next node in list2
            }
            current = current.next; // Move the current pointer forward
        }

        // If there are remaining nodes in either list, link them
        if (list1 != null) {
            current.next = list1;
        } else {
            current.next = list2;
        }

        return dummy.next; // Return the merged list, skipping the dummy node
    }
}
